### PR TITLE
0.5.1

### PR DESCRIPTION
**Pull request by yiwen-h**
Update to package aimed at improving performance for targeting minor categories. Includes additional functions for merging categories together.

API also improved, with much more detailed documentation.

#85

### CODE REVIEW

File: api/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+anyio==3.6.2 ; python_version >= "3.8" and python_version < "3.11"
+astunparse==1.6.3 ; python_version >= "3.8" and python_version < "3.11"
+attrs==22.2.0 ; python_version >= "3.8" and python_version < "3.11"
+blis==0.7.9 ; python_version >= "3.8" and python_version < "3.11"
+cachetools==5.3.0 ; python_version >= "3.8" and python_version < "3.11"
+catalogue==2.0.8 ; python_version >= "3.8" and python_version < "3.11"
+certifi==2022.12.7 ; python_version >= "3.8" and python_version < "3.11"
+chardet==4.0.0 ; python_version >= "3.8" and python_version < "3.11"
+click==8.1.3 ; python_version >= "3.8" and python_version < "3.11"
+colorama==0.4.6 ; python_version >= "3.8" and python_version < "3.11" and sys_platform == "win32" or python_version >= "3.8" and python_version < "3.11" and platform_system == "Windows"
```
Comment (ChrisBeeley) [inline api/requirements.txt:11]: What's this line about? It looks like it won't run on Linux, which is what the server is running. Or am I not understanding?

File: api/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+spacy-legacy==3.0.12 ; python_version >= "3.8" and python_version < "3.11"
+spacy-loggers==1.0.4 ; python_version >= "3.8" and python_version < "3.11"
+spacy==3.5.1 ; python_version >= "3.8" and python_version < "3.11"
+srsly==2.4.6 ; python_version >= "3.8" and python_version < "3.11"
+starlette==0.26.1 ; python_version >= "3.8" and python_version < "3.11"
+tensorboard-data-server==0.7.0 ; python_version >= "3.8" and python_version < "3.11"
+tensorboard-plugin-wit==1.8.1 ; python_version >= "3.8" and python_version < "3.11"
+tensorboard==2.12.0 ; python_version >= "3.8" and python_version < "3.11"
+tensorflow-estimator==2.12.0 ; python_version >= "3.8" and python_version < "3.11"
+tensorflow-io-gcs-filesystem==0.31.0 ; python_version >= "3.8" and python_version < "3.11" and platform_machine != "arm64" or python_version >= "3.8" and python_version < "3.11" and platform_system != "Darwin"
```
Comment (ChrisBeeley) [inline api/requirements.txt:86]: Wow, I don't even know what Darwin is. Same question, is this portable?

File: pxtextmining/factories/factory_pipeline.py [hunk 3]
```diff
@@ -293,7 +281,7 @@ def create_sklearn_pipeline(model_type, tokenizer=None, additional_features=True
                     probability=True,
                     class_weight="balanced",
                     max_iter=1000,
-                    cache_size=500,
+                    cache_size=1000,
                 ),
                 n_jobs=-1,
```
Comment (ChrisBeeley) [inline pxtextmining/factories/factory_pipeline.py:286]: Does this use all cores? May be a problem on the server (or it may not, I'm really not sure)

**Review by ChrisBeeley — COMMENTED** [review]
A few minor points and comments, merge at will

File: api/requirements.txt (hunk 2 above)
Comment (yiwen-h) [reply api/requirements.txt:86]: Tested installing the requirements.txt file in CDU linux server, which went ok.

**Pull request merged**